### DISCUSSION TITLE
Image processing issues in GUI version of vips/nip2  v.8.6.0 (Mac OSX)

### DISCUSSION THREAD

**Discussion by freyaSH** [libvips/libvips#2567] [Q&A]
Hello 

I’m a conservator student trying to learn nip2/vips and I’m having some issues with the GUI based version of NIp2/Vips (8.6.0) for OSX (Monterey). I'm using the "Charisma guide" where the nip2 steps are explained as my walkthrough. Unfortunately, I am not a programmer, so I don’t know what to do about the errors. 

So, I am trying to create a false colour infrared image and all steps go well until I get to the "falsecolour" tab. Some functions have gone missing in the K9 & K2 in the “falsecolour” tab, and in B2 & B4 in the “results” tab, which means that a false colour image can't be made. The functions in these sections become unavailable after having uploaded the VIS and IR images. 

I get the following error messages:

Error in row.
Error in row K9.object: Bad left hand side of '.'.
Error in binary ..
left = no value
right = .value
Called from "Pair_falsecolour.object" (K9.object, K2.object)

Error in row. 
Error in row K2.object: Bad left hand side of '.'.
Error in binary ..
left = no value
right = .value
Called from "Pair_falsecolour.object" (K9.object, K2.object)

Bad left hand side of '.'.
Error in binary ..
left = <function>
right = .K9
Called from "B2" (B2)

Bad left hand side of '.'.
Error in binary ..
left = <function>
right = .K2
Called from "B4" (B4)

Many thanks in advance for any help

**Comment by jcupitt** [comment]
Hi @freyaSH,

Sorry, it's very hard to work out what could have gone wrong from your description. I'll try to make a false-colour infrared later today and post a list of the steps.

**Reply by freyaSH** [reply]
Thank you Jcupitt. I have had a closer look and seem to have found the issue, it is now working. I will post it here later.

**Reply by freyaSH** [reply]
@jcupitt 

So I have now made a little video to demonstrate the errors that stop the FCIR image from compiling; here is the link: https://youtu.be/epSVXHhtbEQ 

**I am using the Charisma guide as my step-by-step walkthrough**

I have put in timestamps at the crucial moments in the workflow that either demonstrate the errors or how the program suddenly fixes itself. I have no idea what is happening at 4:00. I have several times tried clicking away the "compatibility mode." notices and recalculated B2 & B4 as well as K2 & K9, but suddenly the program works. However,  the errors are back every time I start a new project.

**Reply by freyaSH** [reply]
Update: So I did everything from scratch to see whether the sudden fix happened a third time. It didn't unfortunately, so I was stuck with the K2 & K9 as well as the B2 & B4 errors again. But, I then decided to save the workspace and reopen it. Once the workspace was reopened the errors went away, and the generation of FCIR and UVRFC was successful.

**Reply by jcupitt** [reply]
Hi again, thanks for the video, I'll try this over the weekend.
